modules/hivemq-edge-module-http/src/main/java/com/hivemq/edge/adapters/http/config/HttpAdapterConfig.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.edge.adapters.http.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.hivemq.adapter.sdk.api.annotations.ModuleConfigField;
+import com.hivemq.adapter.sdk.api.config.ProtocolAdapterConfig;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static com.hivemq.edge.adapters.http.config.HttpAdapterConfig.HttpContentType.JSON;
+import static com.hivemq.edge.adapters.http.config.HttpAdapterConfig.HttpMethod.GET;
+import static com.hivemq.edge.adapters.http.HttpAdapterConstants.DEFAULT_TIMEOUT_SECONDS;
+import static com.hivemq.edge.adapters.http.HttpAdapterConstants.MAX_TIMEOUT_SECONDS;
+
+@JsonPropertyOrder({"id", "url", "httpConnectTimeout", "httpToMqtt"})
+public class HttpAdapterConfig implements ProtocolAdapterConfig {
+
+    private static final @NotNull String ID_REGEX = "^([a-zA-Z_0-9-_])*$";
+
+    public static final @NotNull String HTML_MIME_TYPE = "text/html";
+    public static final @NotNull String PLAIN_MIME_TYPE = "text/plain";
+    public static final @NotNull String JSON_MIME_TYPE = "application/json";
+    public static final @NotNull String XML_MIME_TYPE = "application/xml";
+    public static final @NotNull String YAML_MIME_TYPE = "application/yaml";
+
+    @JsonProperty(value = "id")
+    @ModuleConfigField(title = "Identifier",
+                       description = "Unique identifier for this protocol adapter",
+                       format = ModuleConfigField.FieldType.IDENTIFIER,
+                       required = true,
+                       stringPattern = ID_REGEX,
+                       stringMinLength = 1,
+                       stringMaxLength = 1024)
+    private final @NotNull String id;
+
+    @JsonProperty("url")
+    @ModuleConfigField(title = "URL",
+                       description = "The url of the HTTP request you would like to make",
+                       format = ModuleConfigField.FieldType.URI,
+                       required = true)
+    private final @NotNull String url;
+
+    @JsonProperty("httpConnectTimeout")
```

Comment:
```suggestion
    @JsonProperty("httpConnectTimeoutSeconds")
```
Now that we are changing the config anyhow.